src/main/kotlin/uk/gov/justice/digital/hmpps/courthearingeventreceiver/service/MessageNotifier.kt

```diff
@@ -34,10 +48,51 @@ class MessageNotifier(
         .stringValue(hearingEventType.description)
         .build()
 
+    when {
+      messageLargerThanThreshold(hearingEvent) -> publishLargeMessage(hearingEventType = hearingEventType, hearingEvent = hearingEvent)
+      else -> publishMessage(hearingEvent, messageTypeValue, hearingEventTypeValue)
+    }
+  }
+
+  private fun publishMessage(hearingEvent: HearingEvent, messageTypeValue: MessageAttributeValue, hearingEventTypeValue: MessageAttributeValue) {
     val publishResult = topic.publish(eventType = "commonplatform.case.received", event = objectMapper.writeValueAsString(hearingEvent), attributes = mapOf("messageType" to messageTypeValue, "hearingEventType" to hearingEventTypeValue), messageGroupId = MESSAGE_GROUP_ID)
+
     log.info("Published message with message Id {}", publishResult.messageId())
```

Comment:
A minor point but this line is duplicated. You could perhaps get the publishMessage functions to return  the messageId and just have one log line.
You could also perhaps remove it - not sure how useful it is